### PR TITLE
Refine logging in pkg/selector/physicalmachine and Refractor functions 

### PR DESCRIPTION
**Pull request by basit9958**
Signed-off-by: Basit Hasan <hasanbasit13@gmail.com>

### What problem does this PR solve?
Refine Logging in `pkg/selector/physicalmachine`
#2887 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/selector/physicalmachine/selector.go [hunk 1]
```diff
@@ -212,7 +210,7 @@ func selectSpecifiedPhysicalMachines(ctx context.Context, c client.Client, spec
 	for ns, names := range spec.PhysicalMachines {
 		if !clusterScoped {
 			if targetNamespace != ns {
-				log.Info("skip namespace because ns is out of scope within namespace scoped mode", "namespace", ns)
+				log.L().WithName("physical-machine-selector").Info("skip namespace because ns is out of scope within namespace scoped mode", "namespace", ns)
```
Comment (iguoyr) [inline pkg/selector/physicalmachine/selector.go:213]: Hi, as the issue #2887 said, please move package level variable `log` as a field of `SelectImpl` struct, but not use the global logger.

File: pkg/selector/physicalmachine/selector.go [hunk 2]
```diff
@@ -238,7 +236,7 @@ func selectSpecifiedPhysicalMachines(ctx context.Context, c client.Client, spec
 			}
 
 			if apierrors.IsNotFound(err) {
-				log.Error(err, "PhysicalMachine is not found", "namespace", ns, "physical machine name", name)
+				log.L().WithName("physical-machine-selector").Error(err, "PhysicalMachine is not found", "namespace", ns, "physical machine name", name)
```
Comment (iguoyr) [inline pkg/selector/physicalmachine/selector.go:239]: ditto

File: pkg/selector/physicalmachine/selector.go (hunk 1 above)
Comment (STRRL) [reply pkg/selector/physicalmachine/selector.go:213]: +1 PTAL @basit9958

Comment (basit9958) [reply pkg/selector/physicalmachine/selector.go:213]: I'm on it

**Comment by STRRL** [comment]
Hi @basit9958 , do you need some help?

**Comment by g1eny0ung** [comment]
Close due to #4037.

**Pull request closed without merging**